source/coll_dist.f90

```diff
@@ -0,0 +1,443 @@
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-04-16
+!  Updated: 2019-04-16
+!  The radial format is just format 1 with a different set of parameters
+! ================================================================================================ !
+subroutine cdist_makeRadial
+
+  real(kind=fPrec) tmpVal(4)
+
+  tmpVal(1)    = cdist_ampX
+  tmpVal(2)    = cdist_ampY
+  tmpVal(3)    = cdist_smearX
+  tmpVal(4)    = cdist_smearY
+
+  cdist_ampX   = cdist_ampR/sqrt(two)
+  cdist_ampY   = cdist_ampR/sqrt(two)
+  cdist_smearX = cdist_smearR/sqrt(two)
+  cdist_smearY = cdist_smearR/sqrt(two)
+
+  call cdist_makeDist_fmt1
+
+  cdist_ampX   = tmpVal(1)
+  cdist_ampY   = tmpVal(2)
+  cdist_smearX = tmpVal(3)
+  cdist_smearY = tmpVal(4)
+
+end subroutine cdist_makeRadial
+
+! ================================================================================================ !
+!  Generation of Distribution Format 1
+!  This routine generates particles in phase space X/XP and Y/YP ellipses, as defined in the input
+!  parameters. Distribution is flat in the band. X and Y are fully uncorrelated.
+! ================================================================================================ !
+subroutine cdist_makeDist_fmt1
+
+  use crcoall
+  use mod_ranlux
+  use mathlib_bouncer
+  use mod_common, only : napx
+  use mod_common_main, only : xv1, xv2, yv1, yv2, ejv, sigmv
+
+  real(kind=fPrec) :: emitX, emitY, sigmaX, sigmaY
+  integer          :: j
+
+  if(cdist_ampX > zero) then
+    do j=1, napx
+      emitX  = cdist_emitX*(cdist_ampX + ((two*(rndm4()-half))*cdist_smearX))**2
+      sigmaX = sqrt(cdist_betaX*emitX)
+      xv1(j) = sigmaX * sin_mb(twopi*rndm4())
+      if(rndm4() > half) then
+        yv1(j) = sqrt(emitX/cdist_betaX-xv1(j)**2/cdist_betaX**2)-(cdist_alphaX*xv1(j))/cdist_betaX
+      else
+        yv1(j) = -one*sqrt(emitX/cdist_betaX-xv1(j)**2/cdist_betaX**2)-(cdist_alphaX*xv1(j))/cdist_betaX
+      end if
+    end do
+  else
+    xv1(:) = zero
+    yv1(:) = zero
+  end if
+
+  if(cdist_ampY > zero) then
+    do j=1, napx
+      emitY  = cdist_emitY*(cdist_ampY + ((two*(rndm4()-half))*cdist_smearY))**2
+      sigmaY = sqrt(cdist_betaY*emitY)
+      xv2(j) = sigmaY * sin_mb(twopi*rndm4())
+      if(rndm4() > half) then
+        yv2(j) = sqrt(emitY/cdist_betaY-xv2(j)**2/cdist_betaY**2)-(cdist_alphaY*xv2(j))/cdist_betaY
+      else
+        yv2(j) = -one*sqrt(emitY/cdist_betaY-xv2(j)**2/cdist_betaY**2)-(cdist_alphaY*xv2(j))/cdist_betaY
+      end if
+    end do
+  else
+    xv2(:) = zero
+    yv2(:) = zero
+  end if
+
+  ejv(:)   = cdist_energy
+  sigmv(:) = zero
+
+end subroutine cdist_makeDist_fmt1
+
+! ================================================================================================ !
+!  Generation of Distribution Format 2
+!  Uses format 1, and generates the transverse beam size in the direction set to zero in fort.3
```

Comment:
very disgusting comment - please, change it saying that the distribution is Gaussian in the other plane
moreover, please mention that format 1 is called outside of this routine, and it is a kind of pre-requisite